### ISSUE TITLE
Weird behaviour when accessing type symbols

### ISSUE THREAD

**Issue by bspot** [dsherret/ts-simple-ast#1171]
**Describe the bug**

Version: 11.0.0 (Typescript 4.3.5)

Please excuse if I'm not explaining this very clearly. I have very little experience with the Typescript compiler and its terminology.

It appears that accessing the symbol of a type changes what the compiler believes the type is / how it resolves aliases.

**To Reproduce**

The following example outputs some information about the properties of the return type of `foo()` in two different ways:

```ts
import { Project } from 'ts-morph'

// Target code
type Jsonify<T> = T extends { toJSON(): infer U }
  ? U
  : T extends object
    ? {
      [k in keyof T]: Jsonify<T[k]>;
    }
    : T;

class Foo {
  a: Date = new Date();

  b!: string[];

  j: Date = new Date();
}

function foo() {
  return {} as Jsonify<Foo>
}

// Test functions
function loadFunction() {
  const project = new Project()
  project.addSourceFileAtPath(__filename)

  const file = project.getSourceFile(__filename)!
  return file.getFunctions()[0]
}

function test1() {
  const fn = loadFunction()
  for (const p of fn.getReturnType().getProperties()) {
    const t = p.getTypeAtLocation(fn)
    console.log(p.getName(), t.getText())
  }
}

function test2() {
  const fn = loadFunction()
  for (const p of fn.getReturnType().getProperties()) {
    const t = p.getTypeAtLocation(fn)
    t.getSymbol()
    console.log(p.getName(), t.getText())
  }
}

console.log("TEST 1")
test1()

console.log()
console.log("TEST 2")
test2()
```

The only difference between `test1()` and `test2()` is that I call `getSymbol()` on the type of the properties in `test2()` but not in `test1()`.

The output is:

```
> ./node_modules/.bin/ts-node ./src/test.ts 
TEST 1
a string true
b string[] false
j string true

TEST 2
a string true
b Array<string> false
j Jsonify<Date> false
```

As you can see, there are multiple differences:

1. The text representation of `b` and `j` is different.
2. The compiler believes that `j` is a string in `test1()` but not in `test2()`
3. Interestingly, the result is consistent for the `a` property.

It appears that in `test1()`, the compiler is resolving the `Jsonify`, but not in `test2()`.

**Expected behavior**

I would expect both functions to produce the same result.

**Comment by dsherret** [comment]
Interesting. This will need to be investigated more. The `getSymbol()` call is essentially only compiler API code. I've tried to reproduce this with only the compiler API (ts-morph bootstrap actually), but I've been unable yet... maybe I've been doing something differently:

```ts
// run with: deno run --unstable file.ts
import { createProjectSync, ts } from "https://deno.land/x/ts_morph@11.0.2/bootstrap/mod.ts";

// Target code
const code = `type Jsonify<T> = T extends { toJSON(): infer U }
  ? U
  : T extends object
    ? {
      [k in keyof T]: Jsonify<T[k]>;
    }
    : T;

class Foo {
  a: Date = new Date();

  b!: string[];

  j: Date = new Date();
}

function foo() {
  return {} as Jsonify<Foo>
}`

const project = createProjectSync({ useInMemoryFileSystem: true })
const file = project.createSourceFile("test.ts", code);
const fn = file.statements[2] as ts.FunctionDeclaration;
const typeChecker = project.createProgram().getTypeChecker();

const fnSignature = typeChecker.getSignatureFromDeclaration(fn)!;
const returnType = fnSignature.getReturnType();
const formatFlags = ts.TypeFormatFlags.UseTypeOfFunction | ts.TypeFormatFlags.NoTruncation | ts.TypeFormatFlags.UseFullyQualifiedType
 | ts.TypeFormatFlags.WriteTypeArgumentsOfSignature;

console.log(typeChecker.typeToString(returnType, undefined, formatFlags));

for (const p of returnType.getProperties()) {
  const t = typeChecker.getTypeOfSymbolAtLocation(p, fn);
  console.log(p.name, typeChecker.typeToString(t, undefined, formatFlags))
  t.getSymbol();
  console.log(p.name, typeChecker.typeToString(t, undefined, formatFlags))
}
```

**Comment by dsherret** [comment]
@bspot fixed in 11.0.3. Thanks for the easily reproducible steps!
